go/api/base_client.go

```diff
@@ -653,6 +653,39 @@ func (client *baseClient) SMIsMember(key string, members []string) ([]Result[boo
 	return handleBooleanArrayResponse(result)
 }
 
+func (client *baseClient) SScan(key string, cursor string) (string, []string, error) {
```

Comment:
Please return `string, []string` wrapped in a `Result` as `Result[string], []Result[string]`.

Reply:
or `Result[string, []string]` (if that possible)?

Reply:
I'm a bit curious as to why we choose to return `[]Result[string]` and not `Result[[]string]`, feels a bit easier to use&test only having to call .Value() on the collection result once compared to doing it on each element, or is there something I'm not aware of that plays into this?

Reply:
In my latest commit I'm returning `Result[string], Result[[]string], err` for sscan and the output looks normal, can still make adjustments if needed

Reply:
I think it is overkill. `Result` just adds nullability to a value type, because `nil` cannot be assigned to a `string`. We use this to return an empty result from a command if it failed, so `res.IsNul() == true` when `err != nil`.